core/src/main/java/org/elasticsearch/index/mapper/DocumentMapperParser.java

```diff
@@ -147,7 +118,7 @@ public DocumentMapperParser(@IndexSettings Settings indexSettings, MapperService
         rootTypeParsers.put(FieldNamesFieldMapper.NAME, new FieldNamesFieldMapper.TypeParser());
         this.rootTypeParsers = unmodifiableMap(rootTypeParsers);
         additionalRootMappers = Collections.emptySortedMap();
-        indexVersionCreated = Version.indexCreated(indexSettings);
+        indexVersionCreated = Version.indexCreated(this.indexSettings);
```

Comment:
another minor thing, for consistency can we use `indexSettings.getIndexVersionCreated()`?